include/appimage/desktop_integration/IntegrationManager.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+// system
+#include <string>
+#include <memory>
+
+// local
+#include <appimage/desktop_integration/Exceptions.h>
+
+namespace appimage {
+    namespace desktop_integration {
+        class IntegrationManager {
+        public:
+            explicit IntegrationManager(const std::string& xdgDataHome = std::string());
+
+            virtual ~IntegrationManager();
+
+            /**
+             * @brief Register an AppImage in the system
+             *
+             * Extract the application main desktop entry, icons and mime type packages. Modifies their content to
+             * properly match the AppImage file location and deploy them into the use XDG_DATA_HOME appending a
+             * prefix to each file. Such prefix is composed as "<vendor id>_<appimage_path_md5>_<old_file_name>"
+             *
+             * @param appImagePath
+             */
+            void registerAppImage(const std::string& appImagePath);
```

Comment:
The `IntegrationManager` must not be bonded to a single AppImage. It's a utility class that abstracts our users from the details of the integration process. This comment will apply better to the `Integrator` class which is the one who actually performs the integration of a single AppImage.

Reply:
There is a functional point in your request, using AppImage instances will validate the files before they get into the IntegrationManager and the user will get an early error notification in case of invalid files.

Change request accepted and applied.